todo_app/app.py

```diff
@@ -1,15 +1,76 @@
-from flask import Flask
 
-from todo_app.flask_config import Config
+from todo_app.data.model import ViewModel
+from flask import Flask, render_template, redirect, request
+# from todo_app.flask_config import Config, TrelloConfig
+from todo_app.data.forms import TodoForm
+from todo_app.data.trello import TrelloUtility
+from todo_app.data.model import ViewModel
 
-app = Flask(__name__)
-app.config.from_object(Config)
+import os
 
 
-@app.route('/')
-def index():
-    return 'Hello World!'
 
+def create_app():
+    app = Flask(__name__)
+
+    app.config.from_object('todo_app.flask_config.Config')
+    
+
+    TRELLO_API_KEY=os.environ.get('TRELLO_API_KEY')
+    TRELLO_TOKEN=os.environ.get('TRELLO_TOKEN')    
+    BOARD_ID=os.environ.get('BOARD_ID')  
+    
+    trello_util = TrelloUtility(TRELLO_API_KEY, TRELLO_TOKEN, BOARD_ID)
+
+    def taskSort(item):
+        return item.status
+
+    @app.route('/', methods=['GET'])
+    def index():
+        form = TodoForm()
+        items = trello_util.get_items()
+        items.sort(reverse=True, key=taskSort)  
+        
+        item_view_model = ViewModel(items)
+        return render_template('index.html', view_model=item_view_model, form=form)
+
+    @app.route('/', methods=['POST'])
+    def index_form():
+        form = TodoForm()
+        if form.validate_on_submit():
+            title = request.form['title']
+            description = request.form['description']
+            trello_util.add_item(title, description)
+
+        return redirect('/')            
+
+    @app.route('/in_progress/<id>', methods=['GET'])
```

Comment:
It's a little odd setting these as a GET method, they might be better as a PUT since we're updating the cards

Reply:
Trello API did not support PUT for this have left it as GET